### ISSUE TITLE
Support for mariadb?

### ISSUE THREAD

**Issue by frank-emrich** [links-lang/links#920]
When releasing the links-mysql package, someone from the opam team pointed out that our links-mysql.opam file lists the following dependencies:
```
depends: [
  "ocaml" {>= "4.08.0"}
  "dune" {>= "1.10.0"}
  "conf-mysql"
  "mysql"
  "links"
]
```
However, depending on both `mysql` and `conf-mysql` is slightly redundant, as `mysql` already depends on `("conf-mariadb" | "conf-mysql")`. Therefore, our dependency on `conf-mysql` essentially just means that we rule out mariadb. Is this on purpose or should we get rid of this restriction?

**Comment by jamescheney** [comment]
This is intended (for now).  

We have not tested with mariadb as far as I know; we can remove this once we have done so and made any changes needed.

This is actually something we might do fairly soon, since mariadb supports some temporal database extensions that @SimonJF was planning to compare with his work.

**Comment by jamescheney** [comment]
Incidentally, our dependency doesn't rule out mariadb, it just means that links-mysql installation won't succeed unless conf-mysql is also installed.  So it just has the effect of saying that while ocaml-mysql requires either mysql or mariadb, links-mysql really requires mysql at least at the moment.  It doesn't stop someone from also having mariadb installed or even from trying to use Links with it, I just don;t want people to think this is a possibility when we haven't tried it ourselves.

**Comment by frank-emrich** [comment]
Good point. What I meant by "ruling out" mariadb was that our current configuration doesn't allow installing Links on a platform that provides `conf-mariadb` but not `conf-mysql`.
But that's more of a theoretical aspect, it is not the case that the former is provided on more platforms/configurations than the latter. Since the update to `conf-mysql` a few days ago (potentially triggered by our Links release failing to build on many platforms), it is actually the other way around.
